### DISCUSSION TITLE
Lesson 4 challenge I keep getting the wrong answer --> Can't solve it

### DISCUSSION THREAD

**Discussion by lamtrinh259** [Cyfrin/foundry-full-course-cu#209] [General]
So for lesson 4 challenge NFT, I thought that the correct answer should be the live feed price as of the date (Jul 1st 2023) of when I'm submitting it. I got the price of 192585567930, so I thought the answer has to be:
1. Either 192585567930
2. or this 1925855679300000000000

But both answers are wrong. Can someone help, please? Below is the part of the code with the solve a challenge function.

`// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {AFoundryCourseChallenge} from "./AFoundryCourseChallenge.sol";
import {AggregatorV3Interface} from "./AggregatorV3Interface.sol";

contract LessonFour is AFoundryCourseChallenge {
    error LessonFour__WrongPrice();

    string private constant LESSON_IMAGE = "ipfs://QmSCoS3N8qFy2D3Tn4vTPKTFAC1TAqSZgg9R4uCb7fZL7Q";
    address private immutable i_priceFeed;

    constructor(address fcn, address priceFeed) AFoundryCourseChallenge(fcn) {
        i_priceFeed = priceFeed;
    }

    function getPrice() external view returns (uint256) {
        (, int256 price,,,) = AggregatorV3Interface(i_priceFeed).latestRoundData();
        return uint256(price);
    }

    function getDecimals() private view returns (uint8) {
        return AggregatorV3Interface(i_priceFeed).decimals();
    }

    function getPriceFeed() external view returns (address) {
        return i_priceFeed;
    }

    /*
     * CALL THIS FUNCTION!
     * 
     * @param priceGuess - Your guess for the price based on the feed. It can be withing a few tokens.
     * @param yourTwitterHandle - Your twitter handle. Can be a blank string.
     */
    function solveChallenge(uint256 priceGuess, string memory yourTwitterHandle) external {
        uint256 actualPrice = getPrice();
        if (getDecimals() == 8) {
            actualPrice = actualPrice * 10e10;
        }
        if (actualPrice + 3e18 > priceGuess && actualPrice - 3e18 < priceGuess) {
            _updateAndRewardSolver(yourTwitterHandle);
        } else {
            revert LessonFour__WrongPrice();
        }
    }`

**Comment by shikhar229169** [comment]
@lamtrinh259, from where you checked the price? Make sure to check the price from the correct pricefeeds contract. You can get the contract address by the function getPriceFeed() of LessonFour contract.

**Reply by lamtrinh259** [reply]
@shikhar229169 thank you so much! I solved it! 

Apparently, I was missing 1 zero! I was reading too fast, and I thought it was just 1e10, but it was **10e10** --> That explained my missing zero!

**Reply by 0xtdey** [reply]
So the correct answer was this -  1925855679300 ?

**Reply by lamtrinh259** [reply]
@tamoghna-dey I don't think it'd work if you just blindly copy my answer (you're welcome to give it a try) without understanding what the question is asking. Follow shikhar229169's advice above and you should get there.

**Reply by 0xtdey** [reply]
I haven't blindly copied your answer, I am just asking if you got it as the answer or not. Price of ETH isn't the same now, anyways it ain't gonna work. And I have already solved it, thanks for your response.
